src/super_gradients/training/models/detection_models/customizable_detector.py

```diff
@@ -0,0 +1,199 @@
+
+        return p3, p4, p5
+
+
+class ThreeHeads(nn.Module):
+    """
+    Apply three heads and combine predictions into the shape expected by SG detection losses
+    """
+    def __init__(self, arch_params: Union[HpmStruct, DictConfig], in_channels: List[int]):
+        super().__init__()
+        arch_params = self._pass_num_classes(arch_params)
+
+        Head1Cls = arch_params.head1.type
+        Head2Cls = arch_params.head2.type
+        Head3Cls = arch_params.head3.type
+
+        self.head1 = Head1Cls(arch_params.head1, in_channels[0])
+        self.head2 = Head2Cls(arch_params.head2, in_channels[1])
+        self.head3 = Head3Cls(arch_params.head3, in_channels[2])
+
+    @staticmethod
+    def _pass_num_classes(arch_params: HpmStruct):
+        for i in range(3):
+            arch_params[f'head{i + 1}'].num_classes = arch_params.num_classes
+        return arch_params
+
+    def forward(self, inputs):
+        p3, p4, p5 = inputs
+        return self.combine_preds([self.head1(p3), self.head2(p4), self.head3(p5)])
+
+    def combine_preds(self, preds):
+        outputs = []
+        outputs_logits = []
+        for output, output_logits in preds:
+            outputs.append(output)
+            outputs_logits.append(output_logits)
+
+        return outputs if self.training else (torch.cat(outputs, 1), outputs_logits)
+
+
+class CustomizableDetector(SgModule):
+    """
+    A customizable detector with backbone -> neck -> heads
+
+    A type of each submodule must be defined explicitly. The defaults are:
+      * backbone - FourStageBackbone
+      * neck - PANNeck
+      * heads - ThreeHeads
+
+    By default, initializes BatchNorm eps to 1e-3, momentum to 0.03 and sets all activations to be inplace
+    """
+
+    def __init__(self, arch_params: Union[HpmStruct, DictConfig], type_mapping: Dict = None):
+        """
+        :param arch_params:
+        :param type_mapping: can be passed to recursively resolve string type names in arch_params to actual types
+        """
+        super().__init__()
+        if type_mapping is not None:
+            recursive_type_name_to_type(arch_params, type_mapping)
+
+        self.arch_params = arch_params
+        BackboneCls = get_param(arch_params.backbone, 'type', FourStageBackbone)
+        NeckCls = get_param(arch_params.neck, 'type', PANNeck)
+        HeadsCls = get_param(arch_params.heads, 'type', ThreeHeads)
+
+        self.backbone = BackboneCls(arch_params.backbone, 3)
+        self.neck = NeckCls(arch_params.neck, self.backbone.out_channels)
+        self.heads = HeadsCls(arch_params.heads, self.neck.out_channels)
+
+        self._initialize_weights(arch_params)
+
+    def forward(self, x):
+        x = self.backbone(x)
+        x = self.neck(x)
+        return self.heads(x)
+
+    def _initialize_weights(self, arch_params: Union[HpmStruct, DictConfig]):
+
+        bn_eps = get_param(arch_params, 'bn_eps', 1e-3)
+        bn_momentum = get_param(arch_params, 'bn_momentum', 0.03)
```

Comment:
Can we include a small comment why set BN params to non-default values? Are they known to work for all backbones/heads?
Maybe some arxiv link, or something to explain the reasoning for these exact values.

Reply:
These are values used all over pytorch implementations of various yolo models, but they aren't mentioned in any paper. I still have doubts whether these defaults should be here, since there is nothing to do with yolo in this class. Can just leave a comment about yolos or set them to default BN layer values. WDYT?

Reply:
I decided to remove the defaults altogether, that is, not to touch BN params unless the yaml asks for it. Two reasons:
* makes more sense to emphasize new values in yolo yamls than to have to set torch defaults for models that don't touch this param;
* Even when nothing is set in a yaml, the modules can define different values on definition, it'd be a bug to always override them with torch defaults 

See the change in e7e36ed